doc/developer-guides/coding_guidelines.rst

```diff
@@ -3110,3 +3100,176 @@ Compliant example::
+   UOS;
+c) The name meaning is simplistic or common, such as vcpu or vm.
+
+Compliant example::
+
+    struct acrn_vcpu {
+            ...
+    };
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       struct vcpu {
+               ...
+       };
+
+
+NC-04: Data structures only used by hypervisor shall be named with prefix 'hv'
+==============================================================================
+
+The data structure types include struct, union, and enum.
+This rule applies to the data structure with all the following properties:
+
+a) The data structure is used by multiple modules;
+b) The corresponding resource is only used by hypervisor;
+c) The name meaning is simplistic or common, such as timer.
+
+Compliant example::
+
+    struct hv_timer {
+            ...
+    };
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       struct timer {
+               ...
+       };
+
+
+NC-05: Data structures only used by one module shall be named with the module name as prefix
+============================================================================================
+
+The data structure types include struct, union, and enum.
+This rule applies to the data structure with all the following properties:
+
+a) The data structure is only used by one module;
+b) The name meaning is simplistic or common, such as context.
+
+Compliant example::
+
+    struct instr_emul_ctxt {
+            ...
+    };
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       struct ctxt {
+               ...
+       };
+
+
+NC-06: Data structures related to hardware resource shall be named with the resource name as suffix
+===================================================================================================
+
+The data structure types include struct, union, and enum.
+For example:
+
+a) The data structure related to register shall be named with suffix 'reg';
+b) The data structure related to segment selector shall be named with suffix
+   'sel'.
+
+Compliant example::
+
+    struct lapic_reg {
+            ...
+    };
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       struct lapic {
+               ...
+       };
+
+
+NC-07: Function pointer shall be named with suffix 'fn'
+=======================================================
+
+Compliant example::
+
+    struct firmware_operations firmware_sbl_ops = {
+            .init = sbl_init_fn,
+            .get_rsdp = sbl_get_rsdp_fn,
+    };
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       struct firmware_operations firmware_sbl_ops = {
+               .init = sbl_init,
+               .get_rsdp = sbl_get_rsdp,
+       };
+
+
+NC-08: A variable name can be shortened from its data structure type name
+=========================================================================
+
+A variable name can be shortened from its data structure type name to avoid the
```

Comment:
I'm not sure if this rule is needed because you have rule ID-08 that already forbids a variable name the same as a type name.  Are there any rules about acceptable shortened names?

Reply:
This rule is more like a suggestion. We don't have the rules about acceptable shortened names.
Maybe I should remove this rule? Since its purpose is to make sure ID-08 is satisfied.